valhalla/mjolnir/landmark_database_builder.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <boost/property_tree/ptree.hpp>
+#include <cstdint>
+#include <iostream>
+#include <list>
+#include <string>
+#include <vector>
+
+#include "baldr/graphconstants.h"
+#include "filesystem.h"
+#include "mjolnir/util.h"
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/geometries.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+#include <geos_c.h>
+
+namespace valhalla {
+namespace mjolnir {
+enum class AccessMode { ReadWriteCreate, ReadOnly, ReadWrite };
+
+struct Landmark {
+  std::string name;
+  std::string type;
+  double lng;
+  double lat;
+};
+
+struct LandmarkDatabase {
+public:
+  LandmarkDatabase(const std::string& db_name, AccessMode access_mode = AccessMode::ReadOnly)
+      : db(nullptr), database(db_name), access_mode_(access_mode) {
+    if (!connect_database()) {
+      LOG_ERROR("cannot connect to database");
+    }
+  }
+  ~LandmarkDatabase() {
+    if (did_inserts && !vacuum_analyze()) {
+      LOG_ERROR("cannot do vacuum and analyze");
+    }
+    close_database();
+  }
+
+  bool insert_landmark(const Landmark& landmark);
+
+  bool get_landmarks_in_bounding_box(std::vector<Landmark>* landmarks,
```

Comment:
this interface is surprising, i would have expected:

```suggestion
  std::vector<Landmark> get_landmarks_in_bounding_box(
```

if its possible that there is a failure we can log error and throw an exception. which actually  brings me to another good point, these methods should all have javadoc on them. have a look at some of the existing classes where they describe the returns and what each argument is and what the function generally does.